include/sacn/universe_discovery.h

```diff
@@ -0,0 +1,252 @@
+ *
+ * sacn_universe_discovery_t my_handle;
+ *
+ * // If you want to specify specific network interfaces to use:
+ * etcpal_error_t create_result = sacn_universe_discovery_create(&my_config, &my_handle, my_netints, NUM_MY_NETINTS);
+ * // Or, if you just want to use all network interfaces:
+ * etcpal_error_t create_result = sacn_universe_discovery_create(&my_config, &my_handle, NULL, 0);
+ * // Check create_result here...
+ * 
+ * // Now the thread is running and your callbacks will handle application-side processing.
+ * 
+ * // What if your network interfaces change? Update my_netints and call this:
+ * etcpal_error_t reset_result = sacn_universe_discovery_reset_networking(my_handle, my_netints, NUM_MY_NETINTS);
+ * // Check reset_result here...
+ * 
+ * // To destroy a universe discovery listener, call this:
+ * sacn_universe_discovery_destroy(my_handle);
+ * 
+ * // During application shutdown, everything can be cleaned up by calling sacn_deinit.
+ * sacn_deinit();
+ * @endcode
+ *
+ * Callback demonstrations:
+ * @code
+ * void my_source_update(sacn_universe_discovery_t handle, const EtcPalUuid* cid, const char* name,
+ *                       const uint16_t* sourced_universes, size_t num_sourced_universes, void* context)
+ * {
+ *   if (cid && name)
+ *   {
+ *     char cid_str[ETCPAL_UUID_STRING_BYTES];
+ *     etcpal_uuid_to_string(cid, cid_str);
+ *     printf("Universe discovery (handle %d): Source %s (name %s) ", handle, cid_str, name);
+ *     if(sourced_universes)
+ *     {
+ *       printf("is active on these universes: ");
+ *       for(size_t i = 0; i < num_sourced_universes; ++i)
+ *         printf("%d ", sourced_universes[i]);
+ *       printf("\n");
+ *     }
+ *     else
+ *     {
+ *       printf("is not active on any universes.\n");
+ *     }
+ *   }
+ * }
+ *
+ * void my_source_expired(sacn_universe_discovery_t handle, const EtcPalUuid* cid, const char* name, void* context)
+ * {
+ *   if (cid && name)
+ *   {
+ *     char cid_str[ETCPAL_UUID_STRING_BYTES];
+ *     etcpal_uuid_to_string(cid, cid_str);
+ *     printf("Universe discovery (handle %d): Source %s (name %s) has expired.\n", handle, cid_str, name);
+ *   }
+ * }
+ *
+ * void my_limit_exceeded(sacn_universe_discovery_t handle, void* context)
+ * {
+ *   printf("Universe discovery (handle %d): Source/universe limit exceeded!\n", handle);
+ * }
+ * @endcode
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** A handle to an sACN universe discovery listener. */
+typedef int sacn_universe_discovery_t;
+/** An invalid sACN universe discovery handle value. */
+#define SACN_UNIVERSE_DISCOVERY_INVALID -1
+
+/**
+ * @brief Constant for "infinite" when listening for sources or universes on a source.
+ *
+ * When using dynamic memory, this constant can be passed in when creating a universe discovery listener.
+ * It represents an infinite number of sources or universes on a source.
+ */
+#define SACN_UNIVERSE_DISCOVERY_INFINITE 0
```

Comment:
When the parameters this is used for are changed to ints, this should probably become -1.

Reply:
I may change this to ints, but I think it's useful to keep this at 0.  Then we get a default of "infinite" for free my memsetting the config structure to 0.  It doesn't make sense to have a 0 limit anyway.